python/ray/tune/trial.py

```diff
@@ -177,6 +177,28 @@ def has_trainable(trainable_name):
         ray.tune.registry.TRAINABLE_CLASS, trainable_name)
 
 
+def _find_newest_ckpt(ckpt_dir):
+    try:
+        full_paths = [
+            os.path.join(ckpt_dir, fname) for fname in os.listdir(ckpt_dir)
+            if fname.startswith("experiment_state") and fname.endswith(".json")
+        ]
+        with open(max(full_paths), "r") as f:
+            runner_state = json.load(f)
+        log_path = runner_state["checkpoints"][-1]["logdir"]
```

Comment:
Instead of relying on the runner_state, can you simply just look for the checkpoint directory? using something like `glob` or `tune.util.recursive_fmatch`?